core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala

```diff
@@ -40,6 +40,7 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
   val consumerCount: Int
   val serverCount: Int
   var logDirCount: Int = 1
+  val producerBufferSize: Long = 1024L * 1024L
```

Comment:
Rather than add a new field and treat as a special case, we could just get the value from `producerConfig` and use that in `createProducer`?